### PR TITLE
Archiving a flow should interrupt all the current runs

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/flows/models.py [hunk 1]
```diff
@@ -1030,6 +1030,9 @@ def archive(self):
         self.is_archived = True
         self.save(update_fields=['is_archived'])
 
+        # any outstanding active runs should interrupted
+        FlowRun.bulk_exit(self.runs.filter(is_active=True), FlowRun.EXIT_TYPE_INTERRUPTED)
```
Comment (nicpottier) [inline temba/flows/models.py]: This probably needs to happen in a celery task otherwise it'll time out on really big flows.

Comment (norkans7) [inline temba/flows/models.py]: Updated

File: temba/flows/models.py [hunk 2]
```diff
@@ -1030,6 +1030,9 @@ def archive(self):
         self.is_archived = True
         self.save(update_fields=['is_archived'])
 
+        from .tasks import interrupt_flow_runs_task
```
Comment (nicpottier) [inline temba/flows/models.py]: Any reason we aren't importing this at the top level?

Comment (norkans7) [inline temba/flows/models.py]: tasks import from models too

File: temba/flows/tests.py [hunk 3]
```diff
@@ -103,6 +103,17 @@ def test_get_unique_name(self):
         self.create_secondary_org()
         self.assertEqual(Flow.get_unique_name(self.org2, "Sheep Poll"), "Sheep Poll")  # different org
 
+    def test_archive_interrupt_runs(self):
+        flow = self.create_flow()
+
+        flow.start([], [self.contact, self.contact2])
+        self.assertEquals(2, flow.runs.filter(exit_type=None).count())
```
Comment (rowanseymour) [inline temba/flows/tests.py]: Minor stylistic point - I think we once agreed to write asserts like:

```
self.assertEqual(actual_value, expected_value)
```

**Pull request merged**